api/controllers/FlairController.js

```diff
@@ -50,17 +50,51 @@ module.exports = {
       }
       var user = await User.findOne(app.user);
       var shortened = app.sub === 'pokemontrades' ? 'ptrades' : 'svex';
-      var relevant_flair = Flairs.makeNewCSSClass(_.get(user, 'flair.' + shortened + '.flair_css_class') || '', app.flair, app.sub);
-      user.flair[shortened].flair_css_class = relevant_flair;
-      await Reddit.setUserFlair(req.user.redToken, user.name, relevant_flair, user.flair[shortened].flair_text, app.sub);
+      var css_flair = Flairs.makeNewCSSClass(_.get(user, 'flair.' + shortened + '.flair_css_class') || '', app.flair, app.sub);
+      user.flair[shortened].flair_css_class = css_flair;
+        
+      // If a ptrades app, update flair_text emoji
+        
+      // Mappings from css_flair to emoji names
+      var flair_mappings = {
+        "default" : ":0:",
+        "gen2" : ":2:",
+        "pokeball" : ":10:",
+        "premierball" : ":20:",
+        "greatball" : ":30:",
+        "ultraball" : ":40:",
+        "luxuryball" : ":50:",
+        "masterball" : ":60:",
+        "dreamball" : ":70:",
+        "cherishball" : ":80:",
+        "ovalcharm" : ":90:",
+        "shinycharm" : ":100:",
+        "gsball" : ":GS:"
+      };
+
+      // Grab flair text and split by spaces to swap out emoji text
+      var flair_text = '';
+      var flair_arr = user.flair[shortened].flair_text.split(' ');
```

Comment:
Will this cause problems if a user's flair doesn't already have an emoji (e.g. if a flair app is approved before a user sets their flair text again)? It seems like it would have the effect of overwriting the first word of the flair text in that case.

Reply:
Resolved this by first checking if the user's flair doesn't have emoji.